src/main/java/com/google/sps/servlets/ElectionServlet.java

```diff
@@ -0,0 +1,100 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+@WebServlet("/election")
+public class ElectionServlet extends HttpServlet {
+
+  private static final String baseURL = "https://www.googleapis.com/civicinfo/v2/elections?key=";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        StringBuilder strBuf = new StringBuilder();  
+        
+        HttpURLConnection conn = null;
+        BufferedReader reader = null;
+
+        try {  
+            URL url = new URL(baseURL + "AIzaSyBlNN_uUrbqY2uRaBpKTD-5Jbw6nk8n0_k");
+            conn = (HttpURLConnection) url.openConnection();  
+
+            conn.setRequestMethod("GET");
+            conn.setRequestProperty("Accept", "application/json");
+            
+            if (conn.getResponseCode() != HttpURLConnection.HTTP_OK) {
+                throw new RuntimeException("HTTP GET Request Failed with Error code : "
+                              + conn.getResponseCode());
+            }
+            
+            // Read the content from the defined connection
+            // Using IO Stream with Buffer for increased efficiency 
+	        reader = new BufferedReader(new InputStreamReader(conn.getInputStream(),"utf-8"));
+            String output = null;  
+
+            while ((output = reader.readLine()) != null) {
+                strBuf.append(output);
+            }                 
+        } catch(MalformedURLException e) {
+            response.setContentType("text/html");
+            response.getWriter().println("URL is incorrectly formatted");
+            return;
+        } catch(IOException e) {  
+            response.setContentType("text/html");
+            response.getWriter().println("Cannot retrieve information from provided URL");
+            return;     
+        } finally {
+            if(reader != null) {
+                try {
+                    reader.close();
+                } catch (IOException e) {
+                    response.setContentType("text/html");
+                    response.getWriter().println("Could not read contents found at URL");
```

Comment:
I don't think this is true right?  If the code hits this block, we would have already read the response and just failed to close the reader.  IMO it's ok to just log the error and return the json result anyways.  

In practicality, I don't expect this block to actually get triggered that much, if at all.

Reply:
I’ll leave a screenshot of the error I receive when I define the URL outside of the try catch. I also wanted to leave it in the function in the case where we retrieve the API key from a separate source

Reply:
![image](https://user-images.githubusercontent.com/38237443/88257364-5ef9f900-cc72-11ea-846a-13fda32d8ee5.png)

Reply:
I don't think this has anything to do with defining the URL outside of the try catch.  I'm talking about choosing to abort and return an error because the reader.close() call fails.

Reply:
Apologies -- I mixed this up with another conversation we were having about the try/catch blocks. Changing to logging instead of returning right now.